### ISSUE TITLE
Keyboard keys don't register occasionally

### ISSUE THREAD

**Issue by redmist77** [schellingb/dosbox-pure#550]
I've been playing Pinball Fantasies lately and I've noticed that when I use the keyboard, sometimes the key presses don't register.  I can't replicate it on-demand but it usually happens at least once if you play for about 10 minutes.  PF uses the shift keys for the flippers.   Gamepads don't seem to be affected.

This might be related to #152 

**_Things I've tried_**

Disable automatic keyboard mapper - same issue.
Different PCs - same issue.
Clean install of various older versions of RetroArch with default settings - same issue.
Tested with DOSBox Core & SVN cores in RetroArch - no problems.

I've tried playing around with various latency settings but no luck.
I'm running on Windows 11.

**Comment by schellingb** [comment]
Hmm, that's weird, if it truly is only happening with certain input drivers then it should be a frontend issue. But you said it doesn't happen in the other DOSBox cores so maybe it's more complicated. I'll try to see if I can reproduce this.

> As a side note, the build that you posted has a video issue where the colors are messed up on the table-selection screen on Pinball Fantasies. This is normal for most versions of DOSBox but setting the video to 'VGA' normally fixes it but not with that build.

@carlosefr If you're available for a bit of testing, can you look into this? This seems like something we might have broken with the recent aspect ratio improvements...

**Comment by redmist77** [comment]
@carlosefr Yeah, it happens with "Game Focus" on or off and with a Hotkey on or off.

**Comment by carlosefr** [comment]
@schellingb I spent a couple of hours digging into the issue with Pinball Fantasies and concluded that it's better to revert commit a1b4968 for now. I've opened PR #552 for this.

There is an older and deeper bug (or combination of bugs) in `vga_draw.cpp`, regardless of a1b4968. One symptom is the Pinball Fantasies table selection screen being misreported as 640x240 under SVGA (`vga.draw.double_scan` is somehow set to `1` under SVGA, when 640x480 certainly isn't double-scanned). The behaviour that a1b4968 tried to fix is another symptom.

I can try giving it another go over the weekend. I think it will take some time to track this down but, if successful, I believe that would mean switching to `vgaonly` for Pinball Fantasies wouldn't be needed anymore. The reason why it doesn't work there is the same reason why a1b4968 broke it under `vgaonly` it seems.

**Comment by schellingb** [comment]
The fix for the issue is included in the latest official RetroArch 1.20 release, so I'm closing this issue.
